### PR TITLE
Make Governance slasher able to remove validator and decrease slash m…

### PR DESCRIPTION
**Pull request by martinvol**
- [ ] fix tests

### Other Changes

* Made isL2 internal (fixed version numbers)
* Moved onlyL1 to modifier in (most) cases

### Related issues

- Fixes https://github.com/celo-org/celo-blockchain-planning/issues/658

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### Documentation

_The set of community facing docs that have been added/modified because of this change_

### CODE REVIEW

File: packages/protocol/contracts-0.8/common/EpochManager.sol [hunk 1]
```diff
@@ -480,6 +480,8 @@ contract EpochManager is
     EpochProcessState storage _epochProcessing = epochProcessing;
 
     for (uint i = 0; i < elected.length; i++) {
+      // TODO here the score should be multiplied by the validator's slashing multiplier
+      // but to save gas the scoreSetters have to manually update the score.
```
Comment (martinvol) [inline packages/protocol/contracts-0.8/common/EpochManager.sol]:
```suggestion
```

File: packages/protocol/contracts/governance/GovernanceSlasher.sol [hunk 2]
```diff
@@ -5,15 +5,37 @@ import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
 import "../common/Initializable.sol";
 import "../common/UsingRegistry.sol";
+import "./interfaces/IValidators.sol";
+import "../../contracts-0.8/common/IsL2Check.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
 
-contract GovernanceSlasher is Ownable, Initializable, UsingRegistry {
+// import {console} from "forge-std/console.sol";
```
Comment (pahor167) [inline packages/protocol/contracts/governance/GovernanceSlasher.sol]: can we remove this ?

File: packages/protocol/contracts/governance/GovernanceSlasher.sol [hunk 3]
```diff
@@ -70,6 +108,55 @@ contract GovernanceSlasher is Ownable, Initializable, UsingRegistry {
...
+      electionLessers,
+      electionGreaters,
+      electionIndices
+    );
+
+    if (group != address(0)) {
+      lockedGold.slash(
+        group,
+        penalty,
+        address(0),
```
Comment (pahor167) [inline packages/protocol/contracts/governance/GovernanceSlasher.sol:143]: Should the reporter be Governance ? or possibly slasherExecuter ?

Comment (martinvol) [reply packages/protocol/contracts/governance/GovernanceSlasher.sol:143]: the reporter is meant to get a reward for in the case of downtime slasher for taking the effort of reporting it, here is a multisig that it's permissioned to do this, so it doesn't make sense IMO to incentivate it to slash people

**Review by pahor167 — APPROVED** [review]

File: packages/protocol/test/common/integration.ts [hunk 4]
```diff
@@ -244,11 +244,11 @@ contract('Integration: Governance slashing', (accounts: string[]) => {
       await governanceSlasher.slash(slashedAccount, lessers, greaters, indices)
     })
 
-    it('should set approved slashing to zero', async () => {
+    it.skip('should set approved slashing to zero', async () => {
```
Comment (martinvol) [inline packages/protocol/test/common/integration.ts]:
```suggestion
    it('should set approved slashing to zero', async () => {
```

File: packages/protocol/test/common/integration.ts [hunk 5]
```diff
@@ -244,11 +244,11 @@ contract('Integration: Governance slashing', (accounts: string[]) => {
...
       assert.equal((await governanceSlasher.getApprovedSlashing(slashedAccount)).toNumber(), 0)
     })
 
-    it('should slash the account', async () => {
+    it.skip('should slash the account', async () => {
```
Comment (martinvol) [inline packages/protocol/test/common/integration.ts]:
```suggestion
    it('should slash the account', async () => {
```

**Pull request merged**